### PR TITLE
Improvements and Error Handling for GitLab Provider

### PR DESCRIPTION
**Pull request by mrT23**
PR Type:
**Bug fix**

___
PR Description:
**This PR introduces several improvements and error handling mechanisms in the GitLab provider. It includes better logging, handling of code suggestions, and error handling in case of exceptions. Also fixed a major bug - additional unneeded line was added to code suggestion in GitLab**

___
PR Main Files Walkthrough:
-`git_providers/gitlab_provider.py`: The changes in this file mainly focus on improving the handling of inline comments and code suggestions. A logging statement has been added for debugging purposes. The code for publishing code suggestions has been wrapped in a try-except block to handle any exceptions that may occur. The logic for handling the 'edit_type' and line numbers for code suggestions has been updated.
-`tools/pr_code_suggestions.py`: The changes in this file focus on improving the handling of inline code suggestions. The code for pushing inline code suggestions has been wrapped in a try-except block to handle any exceptions. The logic for handling 'relevant lines' in the code suggestions has been updated to handle cases where there are multiple ranges.

### CODE REVIEW

**Review by okotek — APPROVED** [review]

**Comment by mrT23** [comment]
Preparing review...

File: pr_agent/git_providers/gitlab_provider.py [hunk 1]
```diff
@@ -140,32 +140,42 @@ def send_inline_comment(self, body, edit_type, found, relevant_file, relevant_li
...
-            body = body.replace('```suggestion', f'```suggestion:-0+{range}')
-
-            lines = target_file.head_file.splitlines()
-            relevant_line_in_file = lines[relevant_lines_start - 1]
-            edit_type, found, source_line_no, target_file, target_line_no = self.find_in_file(target_file,
-                                                                                              relevant_line_in_file)
-            self.send_inline_comment(body, edit_type, found, relevant_file, relevant_line_in_file, source_line_no,
-                                     target_file, target_line_no)
+                        if file.filename == relevant_file:
+                            target_file = file
```
Comment (mrT23) [inline pr_agent/git_providers/gitlab_provider.py:160]: **Suggestion:** Remove the duplicate condition check for `file.filename == relevant_file`.
```suggestion
                for file in self.diff_files:
                    if file.filename == relevant_file:
                        target_file = file
                        break
```

**Comment by mrT23** [comment]
Preparing pr description...

**Comment by mrT23** [comment]
Preparing pr description...

**Comment by mrT23** [comment]
Preparing pr description...

**Comment by mrT23** [comment]
## PR Title

The GitLab Chronicles: A Tale of Error Handling and Code Suggestions

## PR Type

Bug fix with tests

## PR Description

In a world where code suggestions and inline comments are the bread and butter of code review, our hero, the GitLab provider, faces a daunting task. This PR is a thrilling saga of how our hero learns to handle errors with grace and improves the way it deals with inline comments and code suggestions. It's a rollercoaster ride of debugging, coding, and logging. Grab your popcorn, folks!

## PR Main Files Walkthrough

-`git_providers/gitlab_provider.py`: The epicenter of our tale. Our hero learns to log its actions while creating comments, and handles exceptions while publishing code suggestions. It also fixes a bug where an extra line was being added to code suggestions.
-`tools/pr_code_suggestions.py`: Here, our hero learns to handle exceptions while parsing code suggestions and improves the way it handles relevant lines in the suggestions.
-`CHANGELOG.md`: The chronicle of our hero's journey, documenting the improved error handling and logging, better handling of inline comments and code suggestions, and the bug fix.

**Comment by lufficc** [comment]
Preparing pr description...

**Pull request merged**